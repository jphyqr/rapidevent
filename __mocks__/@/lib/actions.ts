'use server'

import { Submission, CustomField, FieldType } from '@/lib/types/customField'

// Mock data with proper types
export const mockData: Submission[] = [
  {
    id: '1',
    name: 'John Doe',
    email: 'john@example.com',
    age: 25,
    customFields: [
      {
        id: 'cf1',
        type: 'text',
        label: 'Company',
        value: 'Acme Inc',
        isAdvanced: false,
        required: false
      } as CustomField
    ],
    createdAt: new Date('2024-01-15T00:00:00.000Z'),
    updatedAt: new Date('2024-01-15T00:00:00.000Z')
  },
  {
    id: '2',
    name: 'Jane Smith',
    email: 'jane@example.com',
    age: 35,
    customFields: [
      {
        id: 'cf2',
        type: 'number',
        label: 'Years of Experience',
        value: 5,
        isAdvanced: false,
        required: false
      } as CustomField
    ],
    createdAt: new Date('2024-02-15T00:00:00.000Z'),
    updatedAt: new Date('2024-02-15T00:00:00.000Z')
  }
]

// Track state for testing
let submissions: Submission[] = [...mockData]

// Mock functions
export async function createSubmission(submission: Submission) {
  // Ensure dates are Date objects
  const newSubmission = {
    ...submission,
    createdAt: new Date(submission.createdAt),
    updatedAt: new Date(submission.updatedAt),
    customFields: submission.customFields.map(field => ({
      ...field,
      isAdvanced: false
    }))
  }

  if (submissions.some(s => s.email === submission.email)) {
    return { success: false, error: 'Email already exists' }
  }
  
  submissions.unshift(newSubmission)
  return { success: true, id: submission.id }
}

export async function checkEmailExists(email: string, excludeId?: string) {
  return submissions.some(item => 
    item.email === email && item.id !== excludeId
  )
}

export async function getSubmissions({
  page = 1,
  pageSize = 10,
  search = '',
  sortBy = 'updatedAt',
  sortDirection = 'desc'
} = {}) {
  let filtered = [...submissions]
  
  if (search) {
    const searchLower = search.toLowerCase()
    filtered = filtered.filter(item =>
      item.name.toLowerCase().includes(searchLower) ||
      item.email.toLowerCase().includes(searchLower) ||
      item.age.toString().includes(search) ||
      item.customFields.some(field => 
        field.label.toLowerCase().includes(searchLower) ||
        String(field.value).toLowerCase().includes(searchLower)
      )
    )
  }

  const total = filtered.length
  const items = filtered.slice((page - 1) * pageSize, page * pageSize)

  return {
    items,
    metadata: {
      currentPage: page,
      pageSize,
      totalItems: total,
      totalPages: Math.ceil(total / pageSize),
      hasNextPage: page < Math.ceil(total / pageSize),
      hasPreviousPage: page > 1
    }
  }
}

export async function updateSubmission(updatedData: Submission) {
  const emailExists = await checkEmailExists(updatedData.email, updatedData.id)
  if (emailExists) {
    return { success: false, error: 'Email already exists' }
  }

  const index = submissions.findIndex(item => item.id === updatedData.id)
  if (index === -1) {
    return { success: false, error: 'Record not found' }
  }

  // Ensure dates are Date objects and custom fields aren't advanced
  submissions[index] = {
    ...updatedData,
    createdAt: new Date(updatedData.createdAt),
    updatedAt: new Date(updatedData.updatedAt),
    customFields: updatedData.customFields.map(field => ({
      ...field,
      isAdvanced: false
    }))
  }

  return { success: true }
}

export async function getSubmissionStats() {
  const timeline = submissions.reduce((acc, submission) => {
    const date = submission.createdAt.toISOString().slice(0, 7)
    const existing = acc.find(a => a.date === date)
    if (existing) existing.count++
    else acc.push({ date, count: 1 })
    return acc
  }, [] as { date: string; count: number }[])

  const ageRanges = {
    '18-24': { min: 18, max: 24 },
    '25-34': { min: 25, max: 34 },
    '35-44': { min: 35, max: 44 },
    '45-54': { min: 45, max: 54 },
    '55+': { min: 55, max: Infinity }
  }

  const ageDistribution = Object.entries(ageRanges).map(([range, { min, max }]) => {
    const count = submissions.filter(s => s.age >= min && s.age <= max).length
    return {
      range,
      count,
      percentage: (count / submissions.length) * 100
    }
  })

  return {
    timeline,
    ageDistribution,
    customFieldUsage: Object.entries(
      submissions.flatMap(s => s.customFields).reduce((acc, field) => {
        acc[field.type] = (acc[field.type] || 0) + 1
        return acc
      }, {} as Record<string, number>)
    ).map(([type, count]) => ({
      type,
      count,
      percentage: (count / submissions.length) * 100
    })),
    totals: {
      totalSubmissions: submissions.length,
      averageAge: Math.round(submissions.reduce((sum, s) => sum + s.age, 0) / submissions.length),
      lastMonthSubmissions: submissions.filter(s => 
        s.createdAt > new Date(Date.now() - 30 * 24 * 60 * 60 * 1000)
      ).length
    }
  }
}

// Test helpers
export function _clearMockData() {
  submissions = [...mockData]
}

export function _getMockData() {
  return [...submissions]
}