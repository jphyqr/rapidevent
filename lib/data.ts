import { type Submission, type CustomField } from './types/customField'

const generateCustomFields = (): CustomField[] => {
  const fields = [
    { label: 'Company', type: 'text' as const, value: 'Acme Inc' },
    { label: 'Phone', type: 'text' as const, value: '555-0123' },
    { label: 'Department', type: 'text' as const, value: 'Engineering' },
    { label: 'Years of Experience', type: 'number' as const, value: 5 },
    { label: 'Work Email', type: 'email' as const, value: 'work@example.com' }
  ]

  // Randomly select 0-3 fields
  const numFields = Math.floor(Math.random() * 4)
  const selectedFields = fields
    .sort(() => Math.random() - 0.5)
    .slice(0, numFields)

  return selectedFields.map(field => ({
    id: crypto.randomUUID(),
    ...field,
    isAdvanced: false,
    required: false
  }))
}

// Generate 50 mock records

export const mockData: Submission[] = Array.from({ length: 500 }, (_, i) => ({
  id: crypto.randomUUID(),
  name: `User ${i + 1}`,
  email: `user${i + 1}@example.com`,
  age: 20 + Math.floor(Math.random() * 40), // Random age between 20-60
  customFields: generateCustomFields(),
  createdAt: new Date(Date.now() - Math.random() * 10000000000), // Random dates within past ~4 months
  updatedAt: new Date(Date.now() - Math.random() * 10000000000)
}))

