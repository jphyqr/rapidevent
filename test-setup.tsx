import { render } from "@testing-library/react"
import React, { useState } from 'react'
import { SubmissionContext } from '@/app/providers/SubmissionProvider'

// Create mock with explicit type
export const mockCreateSubmissionFn = jest.fn().mockImplementation((data) => {

  return Promise.resolve({ success: true, id: '123' })
})

// Mock server actions
jest.mock('@/lib/actions', () => {
  const actual = jest.requireActual('@/lib/actions')
  return {
    ...actual,
    createSubmission: (...args: any[]) => mockCreateSubmissionFn(...args),
    checkEmailExists: () => Promise.resolve(false)
  }
})

// Other mocks
jest.mock('next/cache', () => ({
  revalidatePath: jest.fn()
}))

jest.mock('@/hooks/use-toast', () => ({
  useToast: () => ({
    toast: jest.fn()
  })
}))

// Provider component
function MockSubmissionProvider({ children }: { children: React.ReactNode }) {
  const [newSubmissionId, setNewSubmissionId] = useState<string | null>(null)
  return (
    <SubmissionContext.Provider value={{ newSubmissionId, setNewSubmissionId }}>
      {children}
    </SubmissionContext.Provider>
  )
}

function customRender(ui: React.ReactElement) {
  return render(ui, { wrapper: MockSubmissionProvider })
}

export { customRender }