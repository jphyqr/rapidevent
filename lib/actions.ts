// app/lib/actions.ts
'use server'

import { revalidatePath } from 'next/cache'
import { mockData } from './data'
import { Submission } from './types/customField'

export async function createSubmission(formData: FormData) {
    const submission = {
      id: crypto.randomUUID(),
      name: formData.get('name') as string,
      email: formData.get('email') as string,
      age: Number(formData.get('age')),
      customFields: JSON.parse(formData.get('customFields') as string),
      createdAt: new Date(),
      updatedAt: new Date()
    }
  
    // Add to start of array to maintain sort order
    mockData.unshift(submission)
    
    revalidatePath('/')
    return { success: true, id: submission.id }
  }

export async function getSubmissions() {
    // Sort by date descending
    return mockData.sort((a, b) => 
      new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
    )
  }

  export async function updateSubmission(updatedData: Submission) {
    const index = mockData.findIndex(item => item.id === updatedData.id)
    if (index !== -1) {
      mockData[index] = {
        ...updatedData,
        updatedAt: new Date()
      }
    }
    revalidatePath('/')
    return { success: true }
  }
  