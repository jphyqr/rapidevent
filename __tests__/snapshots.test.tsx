// __tests__/snapshots.test.tsx
import { getSubmissionStats, getSubmissions } from '@/lib/actions'
import { customRender } from '@/test-setup'
import DataForm from '@/app/_components/data-form'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('Snapshot Tests', () => {
  describe('Stats API', () => {
    it('matches stats response structure', async () => {
      const stats = await getSubmissionStats()
      
      // Create a sanitized version with expected types instead of actual values
      const sanitizedStats = {
        ageDistribution: stats.ageDistribution.map(dist => ({
          range: dist.range,
          count: expect.any(Number),
          percentage: expect.any(Number)
        })),
        customFieldUsage: stats.customFieldUsage.map(usage => ({
          type: usage.type,
          count: expect.any(Number),
          percentage: expect.any(Number)
        })),
        timeline: stats.timeline.map(t => ({
          date: expect.any(String),
          count: expect.any(Number)
        })),
        totals: {
          totalSubmissions: expect.any(Number),
          averageAge: expect.any(Number),
          lastMonthSubmissions: expect.any(Number)
        }
      }
      
      expect(sanitizedStats).toMatchSnapshot()
    })
  })

  describe('Pagination API', () => {
    it('matches paginated response structure', async () => {
      const response = await getSubmissions({
        page: 1,
        pageSize: 2
      })

      // Snapshot the structure but not the data
      const sanitizedResponse = {
        metadata: {
          currentPage: response.metadata.currentPage,
          pageSize: response.metadata.pageSize,
          totalItems: expect.any(Number),
          totalPages: expect.any(Number),
          hasNextPage: expect.any(Boolean),
          hasPreviousPage: expect.any(Boolean)
        },
        // Just verify items is an array
        items: expect.any(Array)
      }

      expect(sanitizedResponse).toMatchSnapshot()
    })
  })

  describe('Form Structure', () => {
    const user = userEvent.setup()

    it('matches empty form layout', () => {
      const { container } = customRender(<DataForm />)
      // Remove any dynamic IDs or classes
      const sanitizedHtml = container.innerHTML.replace(/id="[^"]*"/g, 'id="test-id"')
      expect(sanitizedHtml).toMatchSnapshot()
    })

    it('matches form with single custom field', async () => {
      const { container } = customRender(<DataForm />)
      
      // Add a custom field
      await user.click(screen.getByRole('button', { name: /add field/i }))
      
      // Remove any dynamic IDs or classes
      const sanitizedHtml = container.innerHTML.replace(/id="[^"]*"/g, 'id="test-id"')
      expect(sanitizedHtml).toMatchSnapshot()
    })
  })
})