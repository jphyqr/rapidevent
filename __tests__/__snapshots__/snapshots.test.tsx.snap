// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests Form Structure matches empty form layout 1`] = `"<div class="rounded-xl border bg-card text-card-foreground border-none bg-gradient-to-b from-secondary/50 via-background to-background shadow-md"><div class="flex flex-col space-y-1.5 p-6"><div class="tracking-tight text-2xl font-semibold leading-tight text-primary">Submit Entry</div><div class="text-sm text-muted-foreground">Add a new submission with optional custom fields</div></div><div class="p-6 pt-0 space-y-6"><form novalidate="" class="space-y-6"><div class="grid gap-4 md:grid-cols-2"><div class="space-y-2"><label class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-sm font-medium" for=":r0:-form-item">Name</label><input class="flex h-9 w-full rounded-md border border-input px-3 py-1 text-base transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm bg-card/50 shadow-sm" placeholder="Enter name" id="test-id" aria-describedby=":r0:-form-item-description" aria-invalid="test-id" value="" name="name"><div class="min-h-[20px]"></div></div><div class="space-y-2"><label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70" for=":r1:-form-item">Email</label><input class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm" placeholder="Enter email" id="test-id" aria-describedby=":r1:-form-item-description" aria-invalid="test-id" type="email" value="" name="email"><div class="min-h-[20px]"></div></div><div class="space-y-2"><label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70" for=":r2:-form-item">Age</label><div class="relative"><input class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm" data-testid="test-id" id="test-id" aria-describedby=":r2:-form-item-description" aria-invalid="test-id" inputmode="numeric" type="text" value=""></div><div class="min-h-[20px]"></div></div></div><div class="rounded-lg border border-border/50 bg-muted/30"><div class="flex items-center justify-between border-b border-border/50 p-4"><h3 class="text-sm font-medium text-muted-foreground">Custom Fields</h3><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input shadow-sm hover:text-accent-foreground rounded-md px-3 text-xs h-8 bg-background/50 hover:bg-background" type="button"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-plus mr-2 h-3 w-3"><path d="M5 12h14"></path><path d="M12 5v14"></path></svg>Add Field</button></div><div class="flex h-24 items-center justify-center text-sm text-muted-foreground">No custom fields added</div></div><div class="flex justify-end pt-4"><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-primary/90 hover:bg-primary" type="submit">Submit</button></div></form></div></div>"`;

exports[`Snapshot Tests Form Structure matches form with single custom field 1`] = `"<div class="rounded-xl border bg-card text-card-foreground border-none bg-gradient-to-b from-secondary/50 via-background to-background shadow-md"><div class="flex flex-col space-y-1.5 p-6"><div class="tracking-tight text-2xl font-semibold leading-tight text-primary">Submit Entry</div><div class="text-sm text-muted-foreground">Add a new submission with optional custom fields</div></div><div class="p-6 pt-0 space-y-6"><form novalidate="" class="space-y-6"><div class="grid gap-4 md:grid-cols-2"><div class="space-y-2"><label class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-sm font-medium" for=":r3:-form-item">Name</label><input class="flex h-9 w-full rounded-md border border-input px-3 py-1 text-base transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm bg-card/50 shadow-sm" placeholder="Enter name" id="test-id" aria-describedby=":r3:-form-item-description" aria-invalid="test-id" value="" name="name"><div class="min-h-[20px]"></div></div><div class="space-y-2"><label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70" for=":r4:-form-item">Email</label><input class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm" placeholder="Enter email" id="test-id" aria-describedby=":r4:-form-item-description" aria-invalid="test-id" type="email" value="" name="email"><div class="min-h-[20px]"></div></div><div class="space-y-2"><label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70" for=":r5:-form-item">Age</label><div class="relative"><input class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm" data-testid="test-id" id="test-id" aria-describedby=":r5:-form-item-description" aria-invalid="test-id" inputmode="numeric" type="text" value=""></div><div class="min-h-[20px]"></div></div></div><div class="rounded-lg border border-border/50 bg-muted/30"><div class="flex items-center justify-between border-b border-border/50 p-4"><h3 class="text-sm font-medium text-muted-foreground">Custom Fields</h3><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 border border-input shadow-sm hover:text-accent-foreground rounded-md px-3 text-xs h-8 bg-background/50 hover:bg-background" type="button"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-plus mr-2 h-3 w-3"><path d="M5 12h14"></path><path d="M12 5v14"></path></svg>Add Field</button></div><div class="divide-y divide-border/50"><div class="grid grid-cols-[1fr,120px,1fr,auto] items-center gap-3 p-3" style="opacity: 0.507604086495121; background-color: transparent;"><input class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-8" placeholder="Field label" value="Field 1"><button type="button" role="combobox" aria-controls="radix-:r6:" aria-expanded="false" aria-autocomplete="none" dir="ltr" data-state="closed" class="flex w-full items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&amp;>span]:line-clamp-1 h-8"><span style="pointer-events: none;">Text</span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevron-down h-4 w-4 opacity-50" aria-hidden="true"><path d="m6 9 6 6 6-6"></path></svg></button><select aria-hidden="true" tabindex="-1" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"><option value="text" selected="">Text</option><option value="number">Number</option><option value="email">Email</option><option value="currency">Currency</option><option value="file" disabled="">File Upload</option><option value="date" disabled="">Date/Time</option><option value="location" disabled="">Location</option></select><div class="relative"><div class="space-y-2"><div class="flex items-center justify-between"><label class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-sm font-medium">Field 1</label><div class="flex items-center gap-2"></div></div><input class="flex w-full rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-8 transition-colors" data-testid="test-id" placeholder="Enter field 1" type="text" value=""><div class="min-h-[20px]"></div></div></div><div class="flex items-center gap-2 ml-2"><button type="button" role="switch" aria-checked="false" data-state="unchecked" value="on" class="peer inline-flex shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input h-4 w-7" id="test-id"><span data-state="unchecked" class="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"></span></button><input aria-hidden="true" style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;" tabindex="-1" type="checkbox" value="on"><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 hover:bg-accent rounded-md text-xs h-7 w-7 p-0 text-muted-foreground hover:text-destructive" type="button"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-trash2 h-4 w-4"><path d="M3 6h18"></path><path d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"></path><path d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"></path><line x1="10" x2="10" y1="11" y2="17"></line><line x1="14" x2="14" y1="11" y2="17"></line></svg></button></div></div></div></div><div class="flex justify-end pt-4"><button class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&amp;_svg]:pointer-events-none [&amp;_svg]:size-4 [&amp;_svg]:shrink-0 text-primary-foreground shadow h-9 px-4 py-2 bg-primary/90 hover:bg-primary" type="submit">Submit</button></div></form></div></div>"`;

exports[`Snapshot Tests Pagination API matches paginated response structure 1`] = `
{
  "items": Any<Array>,
  "metadata": {
    "currentPage": 1,
    "hasNextPage": Any<Boolean>,
    "hasPreviousPage": Any<Boolean>,
    "pageSize": 2,
    "totalItems": Any<Number>,
    "totalPages": Any<Number>,
  },
}
`;

exports[`Snapshot Tests Stats API matches stats response structure 1`] = `
{
  "ageDistribution": [
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "range": "18-24",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "range": "25-34",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "range": "35-44",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "range": "45-54",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "range": "55+",
    },
  ],
  "customFieldUsage": [
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "type": "text",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "type": "email",
    },
    {
      "count": Any<Number>,
      "percentage": Any<Number>,
      "type": "number",
    },
  ],
  "timeline": [
    {
      "count": Any<Number>,
      "date": Any<String>,
    },
    {
      "count": Any<Number>,
      "date": Any<String>,
    },
    {
      "count": Any<Number>,
      "date": Any<String>,
    },
    {
      "count": Any<Number>,
      "date": Any<String>,
    },
    {
      "count": Any<Number>,
      "date": Any<String>,
    },
  ],
  "totals": {
    "averageAge": Any<Number>,
    "lastMonthSubmissions": Any<Number>,
    "totalSubmissions": Any<Number>,
  },
}
`;
